Modules/+Experiments/@Saturation/run.m

```diff
@@ -1,39 +1,77 @@
+    obj.meta.angles = obj.angle_list; %Angles corresponding to each spectrum
 
-% Flip the acquire boolean to true, set PM wavelength, and initialize data structures
-obj.acquire = true;
-obj.PM100.set_wavelength(obj.wavelength);
-obj.pm_data = [];
-obj.apd_data = [];
+    % Check that rot is not empty and valid
+    assert(~isempty(obj.rot) && isvalid(obj.rot),'Motor SN must be a valid number; motor handle may have been deleted. Check motor serial number/APT Config')
 
-% Continually collect and plot PM and APD data until the user hits "Stop acquisition", which flips
-% obj.acquire to false
-while obj.acquire
-    obj.pm_data = [obj.pm_data, obj.PM100.get_power('MW')];
-    obj.apd_data = [obj.apd_data, obj.counter.singleShot(obj.counter.dwell, obj.nsamples)];
-    [~, sort_index] = sort(obj.pm_data);
-    plot(ax(1),obj.pm_data(sort_index),obj.apd_data(sort_index))
-    xlabel(ax(1),'Input Power (mW)')
-    ylabel(ax(1), 'Output Counts')
-    plot(ax(2),obj.pm_data)
-    ylabel(ax(2),'Input Power (mW)')
-    xlabel(ax(2), 'Collection Bins')
+    % Instantiate data object
+    Nangles = length(obj.angle_list);
+    obj.data.intensity = nan(1,Nangles);
+    
+    % Setup graphics
+    y = NaN(1,Nangles);
+    hold(ax,'on');
+    plotH(1) = plot(obj.angle_list, y,'color', 'k','parent',ax);
+    ylabel(ax,'Counts (cps)');
+    xlabel(ax,['Angle (' char(176) ')']);
+    yyaxis(ax, 'left');
+
+    try
+        % Home rotation mount
+
+        if ~obj.rot.Homed
+            status.String = 'Homing motor'; drawnow;
+            
+            obj.rot.home();
+            pause4Move(obj, obj.motor_home_time);
+        end
+        
+        % Sweep through polarisation and get spectra
+        for i = 1:Nangles
+            theta = obj.angle_list(i);
+            if ~isempty(obj.rot)
+                status.String = sprintf( 'Navigating to %g (%i/%i)', theta, ...
+                    i, Nangles); drawnow;
+                obj.rot.move(theta);
+            else
+                pause(5)
+            end
+            pause4Move(obj, obj.motor_move_time);
+            status.String = sprintf( 'Measuring at %g (%i/%i)', theta, ...
+                i, Nangles); drawnow;
+            
+            % Measure count rate
+            obj.data.intensity(i) = ctr.singleShot(obj.exposure, 1);
+
+            plotH(1).YData = obj.data.intensity;
+            drawnow; assert(~obj.abort_request,'User aborted');
+        end
+        
+    catch err
+    end
+    % CLEAN UP CODE %
+    if exist('err','var')
+        % HANDLE ERROR CODE %
+        rethrow(err)
+    end
 end
-set(statusH, 'string', 'Acquisition complete!');
 
-% Set "abort" status if the data structures are empty
-if isempty(obj.pm_data) && isempty(obj.apd_data)
-    set(statusH,'string','Aborted!');
+% Wait until motor stops moving, or timeout
+function pause4Move(obj,maxTime)
+    t = tic;
+    while (obj.rot.Moving || ~obj.rot.Homed) && (toc(t) < maxTime)
+        drawnow
+        if toc(t) > maxTime
```

Comment:
The chance that line 73 executes is extremely tiny (`toc(t)` would have had to go from `< maxTime` to `> maxTime` with only a `drawnow` between)! Your while loop is set to exit if maxTime is exceeded without erroring. This seems problematic?